### ISSUE TITLE
Appears to play via WAN for LAN activity

### ISSUE THREAD

**Issue by pbathuk** [plexinc/plex-for-kodi#157]
Hi,

I am using the latest stable version 0.13, Kodi is 17.6 (windows store version) & my server is latest beta version (for the collection goodness).

I have been trying to figure out a way around the "can't play" issue that appears to happen with this version of Plex & after changing some network settings (IPv6 off etc) it seems to work.

Although everything is playing correctly when i check what IP address / route the direct play (99% of my videos are direct play) seem show that is coming from WAN and not the LAN.
When I then look at network panel from task manager it shows the packets are coming from the external IP, looking at the sever shows the packets sent to the external IP as well.

When I use the desktop app or android etc on the LAN it shows as local, so seems to be a Plex for Kodi only issue.

I can load some logs (once the kids have stopped watching frozen... Again)
Cheers

**Comment by pbathuk** [comment]
To add to this, I've have been trying different settings & found that if I turn off remote access on the server it then switches to LAN & the loading of files is much quicker... Therefore it seems to be a logic piece around selecting the connection (remote or local): 
Image to prove that it now shows LAN

![screenshot_20180210-102449 2](https://user-images.githubusercontent.com/25253860/36061105-5fc8e2e6-0e4d-11e8-8f15-ba61c215f323.jpg)

**Comment by pbathuk** [comment]
I have tried the latest server & have re-enabled remote access (as require the feature) and can confirm that the issue is still there. 
I have tried other Plex clients & they connect directly without using the WAN connection, therefore it does seem to be a Plex for Kodi issue only

**Comment by pbathuk** [comment]
I have done some digging through the code & worked out why it is happening, but can't seem to create a fix for it:
Within the plexnet area there are calls to plex.tv to grab the list of connections allowed, as part of the first call it forces Https and ONLY uses this list, therefore any manual connections / local servers are ignored until this list is processed, specifically the server list.
Since plex.tv returns secure connections, they both need to route external to the network & then back in, the local address sent back is not usable internally (192.168... but with the direct.plex.tv atteched)

Therefore the way to "fix" this is to priorities the internal network ahead of plex.tv and then fall back to plex.tv rather than the other way around.

I just can't figure out how to do this

**Comment by pbathuk** [comment]
So I'm glad to report that I do have a working fix on my own code, currently it forces the server to use the local version only when encryption is set to preferred and the client is set to local only (or always). 
I am going to try to tidy it up a little before doing a PR. 
As you can see below, no other settings changed & it's using the LAN not WAN.

This also appears to fix the file not found issue as well.

![screenshot_20180325-111933](https://user-images.githubusercontent.com/25253860/37873992-f4a6ee24-301e-11e8-8153-c77b416a6bd3.jpg)

**Comment by ruuk** [comment]
Hi pbathuk,

Just to clarify one thing here, addresses in the form of `192-168-1-3.blahblahblahblah.plex.direct:32400` are local connections.
If you do a DNS lookup or ping the address you will see that it resolves to the local address (in this example 192.168.1.3). When the app uses these addresses DNS resolves them to the local address so the connection is made locally. I just double checked this now, and PMS logs show that the connection is coming from the local address when using these URLs.
Much of the behavior in this app is modeled on the Roku client, and I can confirm that (at least for me) both the Roku client and Plex for Kodi are using the same URLs when connecting to local servers.
These URLs are required for secure connections, and the app is going to prefer secure local connections over insecure ones.

Now, all that being said, if the URLs being used in your case are not actually local (i.e. they are of the form above, but without a prefix matching a local address) then something else is going on.

**Comment by pbathuk** [comment]
So... I managed to have a few more hours looking at this and trying to figure it out. 
Turns out the BT hub that I have does not allow DNS rebinding, therefore the local address does not reach anywhere, if I try using a different DNS (8.8.8.8) and ntlookup through the command line it works, therefore it is actually a router fault. 
Unfortunately the DNS server can't be changed on the router, therefore I have completed a different work around of adding a line to the hosts file of the pcs impacted to auto resolve the IP address.

Therefore this is not an issue of Plex for Kodi and I will close it. 
FYI the issue of files not found IS the same issue, however I am not sure why, it would appear that because the connection is being made as remote (as a work around to no DNS rebinding) then the remote connection limits are imposed, which makes sense, however the script is not correctly enabling transcoding etc and because they are outside these limits the PMS is not allowing the file to be streamed. I believe a profile on PMS will fix that one, but not sure.
I'm happy to do some digging into the PMS logs and Kodi logs when I get a chance
